packages/actor-query-operation-group/lib/ActorQueryOperationGroup.ts

```diff
@@ -0,0 +1,131 @@
+import { ArrayIterator } from 'asynciterator';
+import { Map, Set } from "immutable";
+import { termToString } from 'rdf-string';
+import { Algebra } from "sparqlalgebrajs";
+import { AggregateEvaluator } from 'sparqlee';
+
+import {
+  ActorQueryOperation,
+  ActorQueryOperationTypedMediated,
+  Bindings,
+  IActorQueryOperationOutputBindings,
+  IActorQueryOperationTypedMediatedArgs,
+} from "@comunica/bus-query-operation";
+import { ActionContext, IActorTest } from "@comunica/core";
+
+/**
+ * A comunica Group Query Operation Actor.
+ */
+export class ActorQueryOperationGroup extends ActorQueryOperationTypedMediated<Algebra.Group> {
+
+  constructor(args: IActorQueryOperationTypedMediatedArgs) {
+    super(args, 'group');
+  }
+
+  public async testOperation(pattern: Algebra.Group, context: ActionContext): Promise<IActorTest> {
+    pattern.aggregates.forEach((aggregate) => {
+      if (aggregate.distinct) {
+        throw new Error("Group Operation doesn't support distinct just yet.");
+      }
+    });
+    return true;
+  }
+
+  public async runOperation(pattern: Algebra.Group, context: ActionContext)
+    : Promise<IActorQueryOperationOutputBindings> {
+    // Get result stream for the input query
+    const { input, aggregates } = pattern;
+    const outputRaw = await this.mediatorQueryOperation.mediate({ operation: input, context });
+    const output = ActorQueryOperation.getSafeBindings(outputRaw);
+    ActorQueryOperation.validateQueryOutput(output, 'bindings');
+
+    // The variables in scope are the variables on which we group, e.g. pattern.variables
+    // for 'GROUP BY ?x, ?z', this is [?x, ?z], for 'GROUP by expr(?x) as ?e' this is [?e].
+    // But also in scope are the variables defined by the aggregations, since GROUP has to handle this
+    const variables = pattern.variables
+      .map((variable) => termToString(variable))
+      .concat(aggregates.map((agg) => termToString(agg.variable)));
+
+    const patternVariables = Set(pattern.variables.map((v) => termToString(v)));
+    const aggregateVariables = Set(aggregates.map(({ variable }) => termToString(variable)));
```

Comment:
Note that immutable.js causes some overhead due to its immutability.
AFAICS, immutability is not needed, so I would suggest using POJO's here.

These Sets can for example just become `{[term; string]: boolean}`, and membership tests can be done using `in`.